src/command_modules/azure-cli-component/azure/cli/command_modules/component/custom.py

```diff
@@ -16,6 +16,10 @@
 COMPONENT_PREFIX = 'azure-cli-'
 
 
+def _deprecate_warning():
+    logger.warning("The 'component' commands will be deprecated in the future.")
```

Comment:
Consider using too the official DeprecationWarning:
https://docs.python.org/2/library/exceptions.html#exceptions.DeprecationWarning
https://docs.python.org/2/library/warnings.html

```python
import warnings
warnings.warn("Method myoperation is deprecated", DeprecationWarning)
```